frontend/providers/applaunchpad/src/pages/api/platform/authDomainChallenge.ts

```diff
@@ -25,22 +27,62 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse)
         error: 'Missing required parameter: customDomain'
       });
     }
+
+    const ip4 = await queryA(customDomain).catch((e) => {
+      console.error('Failed to resolve IPv4 address:', e);
+      return null;
+    });
+    const ip6 = await queryAAAA(customDomain).catch((e) => {
+      console.error('Failed to resolve IPv6 address:', e);
+      return null;
+    });
+
     const token = nanoid();
 
-    const challengeUrl = `http://${customDomain}/api/.well-known/applaunchpad-domain-challenge/${token}`;
+    const challengeUrls = [ip4?.data, ip6?.data, customDomain].flatMap((host) => {
+      if (!host) return [];
+      return [
+        `https://${host}/api/.well-known/applaunchpad-domain-challenge/${token}`,
```

Comment:
There's no need to use HTTPS, as most HTTPS certificates do not include IP addresses.